test_conformance/gl/test_images_write_common.cpp

```diff
@@ -37,8 +37,7 @@ static const char *kernelpattern_image_write_1D =
 "}\n";
 
 static const char *kernelpattern_image_write_1D_half =
-"#pragma OPENCL EXTENSION cl_khr_fp16 : enable\n"
-"__kernel void sample_test( __global half4 *source, write_only image1d_t dest )\n"
+"__kernel void sample_test( __global half *source, write_only image1d_t dest )\n"
 "{\n"
 "    uint index = get_global_id(0);\n"
 "    write_imagef( dest, index, vload_half4(index, (__global half *)source));\n"
```

Comment:
I think we can get rid of all of these casts to `__global half*` now.  For example:

```suggestion
"    write_imagef( dest, index, vload_half4(index, source));\n"
```

Reply:
> I think we can get rid of all of these casts to `__global half*` now. For example:

@bashbaug Of course I will remove such necessary casting.